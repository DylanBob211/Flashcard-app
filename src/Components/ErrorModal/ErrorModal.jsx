import React from 'react';
import PropTypes from 'prop-types';

const ErrorModal = ({ text }) => (
  text.length > 0
    ? (
      <div style={{ 
        backgroundColor: 'rgba(255, 0, 0, 0.5)',
        position: 'absolute',
        left: '50%',
        top: '2%',
        transform: 'translateX(-50%)',
        textAlign: 'center',
        padding: '1rem',
        borderRadius: '5px',
      }}
      >
        <h4>{text}</h4>
      </div>
    )
    : null
);

export default ErrorModal;

ErrorModal.propTypes = {
  text: PropTypes.string.isRequired,
};
