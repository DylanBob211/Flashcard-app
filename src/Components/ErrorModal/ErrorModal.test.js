import React from 'react';
import { mount, shallow } from 'enzyme';
import ErrorModal from './ErrorModal';

describe('ErrorModal', () => {
  let mountedErrorModal;

  const props = {
    text: '',
  };
  const errorModal = ({ text }, isShallow = false) => {
    if (isShallow) return shallow(<ErrorModal text={text} />);
    return mount(<ErrorModal text={text} />);
  };

  beforeEach(() => {
    mountedErrorModal = undefined;
  });

  it('matches to snapshot', () => {
    expect(errorModal(props, true)).toMatchSnapshot();
  });

  it('renders null if text prop is not passed', () => {
    mountedErrorModal = errorModal(props);
    expect(mountedErrorModal.isEmptyRender()).toBeTruthy();
  });

  it('renders markup with the text prop when it is passed', () => {
    props.text = 'hello Error';
    mountedErrorModal = errorModal(props);
    expect(mountedErrorModal.text()).toEqual('hello Error');
  });
});
