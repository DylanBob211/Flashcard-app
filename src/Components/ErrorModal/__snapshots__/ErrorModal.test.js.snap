// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorModal matches to snapshot 1`] = `""`;
